### ISSUE TITLE
Cannot select multiple radio buttons in same control group

### ISSUE THREAD

**Issue by cur3n4** [angular/angular#7051]
In beta.6 when having multiple radio buttons in the same control group, as soon as one is selected all the others in that same control group get updated to unchecked, even though they have a different name.
The issue seems related to this line in RadioControlRegistry:
```
      if (c[0].control.root === accessor._control.control.root && c[1] !== accessor) {
        c[1].fireUncheck();
      }
```
Shouldn't it check that they match the name?
```
      if (c[0].control.root === accessor._control.control.root 
        && c[1] !== accessor && c[1].name === accessor.name) {
        c[1].fireUncheck();
      }
```
Thanks

**Comment by draconisNoctis** [comment]
Same issue here, plunker for example:
https://plnkr.co/edit/wh1TW9JAej4ciCK66TUT

**Comment by cur3n4** [comment]
Also coming back to this, I think it is probably necessary to check the native element name attribute. Something as tacky as this:

```
      if (c[0].control.root === accessor._control.control.root 
        && c[1] !== accessor 
        && c[1]._elementRef.nativeElement.name === (<any>accessor)._elementRef.nativeElement.name) {
        c[1].fireUncheck();
      }

**Comment by PascalPrecht** [comment]
Thanks for reporting this!

**Comment by PascalPrecht** [comment]
Actually, on a second thought, isn't this the behaviour of radio buttons?

You can only make **one** choice of multiple radio buttons (with the same name). For multiple selections you either need to add different `name`'s, or use check boxes.

@draconisNoctis you plunk shows expected behaviour.

Or am I missing something?

**Comment by PascalPrecht** [comment]
Ah nevermind! Seeing it now.

**Comment by mdymel** [comment]
@cur3n4 Did you find some temporary solution to this?

**Comment by cur3n4** [comment]
@mdymel To get around it, I am injecting a custom implementation of RadioControlRegistry

```
import {RadioControlRegistry, RadioControlValueAccessor} from 'angular2/src/common/forms/directives/radio_control_value_accessor';

// TODO Monkey patching Radio buttons until https://github.com/angular/angular/issues/7051 is fixed
export class CustomRadioControlRegistry extends RadioControlRegistry {

    select(accessor: RadioControlValueAccessor) {
        console.log('CustomRadioControlRegistry');
        (<any>this)._accessors.forEach((c) => {
            if (c[0].control.root === accessor._control.control.root
              && c[1] !== accessor && c[1]._elementRef.nativeElement.name === (<any>accessor)._elementRef.nativeElement.name) {
              c[1].fireUncheck();
            }
        });
    }
}
```

And then when bootstrapping the app, I am replacing the RadioControlRegistry with my implementation:
`provide(RadioControlRegistry, {useClass: CustomRadioControlRegistry})`

**Comment by mdymel** [comment]
@cur3n4 thanks! Works like a charm

**Comment by gerhardboer** [comment]
I stumbled upon this issue while trying to find the cause for an issue I submitted: https://github.com/angular/angular/issues/7068.

tl;dr: The number of radios in a form is equal to the number of calls to form.valueChanges.
When you edit the [plunkr](https://plnkr.co/edit/wh1TW9JAej4ciCK66TUT) of @draconisNoctis with a counter, you'll see the result.

I was just looking at the code, and this is exactly the place where the events are thrown from the Radio, but the checks are if they are in the same group, and then fires an event for all radios.

Wouldn't it be preferable to just fire the event of the radio that was clicked on? for text fields, it works this way.
The following addition to the check should do the trick:

`&& c[0].name === accessor._control.name`

Or am i misinterpreting the desired behaviour of the radios? @PascalPrecht

**Comment by rapropos** [comment]
I just banged into this issue as well, and my first instinct was to do exactly what the OP did. @cur3n4 : can you elaborate on why you decided it was necessary to hit the DOM instead of relying on the name property as you did in your original comment?

**Comment by cur3n4** [comment]
@rapropos I can't remember right now but I think it was related to creating checkboxes with dynamic names inside a loop ...

**Comment by rapropos** [comment]
Hmm, I'm not doing that, so maybe I'll be OK relying on that name property. I would also like to suggest that this issue transcends control groups. The way I see it, using `root` as the discriminator will affect all radio buttons in the same form. My first attempt at working around this was to create a separate subgroup for each desired radio group, i.e, instead of:

```
this.form = fb.group({
  ...
  // these two have name="a" on their corresponding HTML elements
   'a1': [new RadioButtonState(false, 'a1')],
   'a2': [new RadioButtonState(false, 'a2')],
...
  // these two have name="b"
   'b1': [new RadioButtonState(false, 'b1')],
   'b2': [new RadioButtonState(false, 'b2')],
...
}
```
I tried:
```
this.form = fb.group({
  ...
  'a': fb.group({
   'a1': [new RadioButtonState(false, 'a1')],
   'a2': [new RadioButtonState(false, 'a2')],
}),
...
'b': fb.group({
   'b1': [new RadioButtonState(false, 'b1')],
   'b2': [new RadioButtonState(false, 'b2')],
})
...
}
```

and changing the corresponding `ngFormControl` bindings to add the additional level of indirection. This didn't help. Selecting any radio button in either group deselected all other buttons, even the ones in other groups.

**Comment by frantisekjandos** [comment]
The bug still remains in 2.0.0-rc.1 :(

**Comment by Paladinium** [comment]
@cur3n4 do you have any idea how to make your workaround compatible with the rc.1 release?
